admin-dev/themes/new-theme/js/pages/order/view/order-prices-refresher.js

```diff
@@ -44,4 +44,42 @@ export default class OrderPricesRefresher {
         $(OrderViewPageMap.orderTaxesTotal).text(response.taxesTotalFormatted);
       });
   }
+
+  refreshProductPrices(orderId) {
+    $.ajax(this.router.generate('admin_orders_product_prices', {orderId}))
+      .then((productPricesList) => {
+        productPricesList.forEach((productPrices) => {
+          const orderProductTrId = '#orderProduct_' + productPrices.orderDetailId;
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditUnitPrice}`).text(productPrices.unitPrice);
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditQuantity}`).text(productPrices.quantity);
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditAvailableQuantity}`).text(productPrices.availableQuantity);
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditTotalPrice}`).text(productPrices.totalPrice);
+
+          // update order row price values
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditBtn}`).data('product-price-tax-incl', productPrices.unitPriceTaxInclRaw);
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditBtn}`).data('product-price-tax-excl', productPrices.unitPriceTaxExclRaw);
+          $(`${orderProductTrId} ${OrderViewPageMap.productEditBtn}`).data('product-quantity', productPrices.quantity);
+        });
+      });
+  }
+
+  checkOtherProductPricesMatch(givenPrice, productId, combinationId) {
+    const productRows = document.querySelectorAll('tr.cellProduct');
+    let unmatchingPriceExists = false;
+
+    productRows.forEach((product) => {
+      const productEditBtn = document.querySelector(`#${product.id} ${OrderViewPageMap.productEditBtn}`);
```

Comment:
This should be in the `OrderViewPageMap` you can define some dynamic selectors with arrow functions, like this
```
productsTableRow: (productId) => `#orderProduct_${productId}`,
```

This would improve the readability a lot, besides you could use jquery like in the rest of the code `const productEditBtn = $(OrderViewPageMap.productEditBtn(product.id))` (maybe the current selector `productEditBtn` should be renamed into `productEditButtons` since it selects multiple buttons, or we just add a new one for the dynamic selector)

Which ultimately allows to get the data as usual `productEditBtn.data('combinationId')` instead of using the dataset Don't get me wrong, everything you did work! It's just to (try ^^) remain consistent with the rest of the code